creusot-contracts/src/lib.rs

```diff
@@ -1,19 +1,124 @@
+    /// The variant must be an expression which returns a type implementing [WellFounded]
+    pub use creusot_contracts_proc::variant;
+
+    /// Enables Pearlite syntax, granting access to Pearlite specific operators and syntax
+    pub use creusot_contracts_proc::pearlite;
+}
+
+#[cfg(not(feature = "contracts"))]
+mod macros {
+    /// A pre-condition of a function or trait item
+    pub use creusot_contracts_dummy::requires;
+
+    /// A post-condition of a function or trait item
+    pub use creusot_contracts_dummy::ensures;
+
+    /// A loop invariant
+    /// The first argument should be a name for the invariant
+    /// The second argument is the Pearlite expression for the loop invariant
+    pub use creusot_contracts_dummy::invariant;
+
+    /// Declares a trait item as being a law which is autoloaded as soon another
+    /// trait item is used in a function
+    pub use creusot_contracts_dummy::law;
+
+    /// Declare a function as being a logical function, this declaration must be pure and
+    /// total. It cannot be called from Rust programs as it is *ghost*, in exchange it can
+    /// use logical operations and syntax with the help of the [pearlite] macro.
+    pub use creusot_contracts_dummy::logic;
+
+    /// Declare a function as being a logical function, this declaration must be pure and
+    /// total. It cannot be called from Rust programs as it is *ghost*, in exchange it can
+    /// use logical operations and syntax with the help of the [pearlite] macro.
+    pub use creusot_contracts_dummy::predicate;
+
+    /// Inserts a *logical* assertion into the code. This assertion will not be checked at runtime
+    /// but only during proofs. However, it has access to the ghost context and can use logical operations
+    /// and syntax.
+    pub use creusot_contracts_dummy::proof_assert;
+
+    /// Instructs Pearlite to ignore the body of a declaration, assuming any contract the declaration has is
+    /// valid.
+    pub use creusot_contracts_dummy::trusted;
+
+    /// Declares a variant for a function, this is primarily used in combination with logical functions
+    /// The variant must be an expression which returns a type implementing [WellFounded]
+    pub use creusot_contracts_dummy::variant;
+
+    /// Enables Pearlite syntax, granting access to Pearlite specific operators and syntax
+    pub use creusot_contracts_proc::pearlite;
+}
+
+pub use macros::*;
 
 #[cfg(feature = "contracts")]
 pub mod stubs;
 
 #[cfg(feature = "contracts")]
 pub mod logic;
 
-#[cfg(feature = "contracts")]
-pub use logic::*;
-
 #[cfg(feature = "contracts")]
 pub mod std;
 
+#[cfg(not(feature = "contracts"))]
+pub mod std {
+    pub use std::vec;
+}
+
+#[cfg(not(feature = "contracts"))]
+pub mod logic {
+    pub struct Ghost<T>(std::marker::PhantomData<T>)
+    where
+        T: ?Sized;
+
+    impl<T> Ghost<T> {
+        pub fn record(_: &T) -> Ghost<T> {
```

Comment:
Since `Ghost` appears in Rust programs, I added a dummy implementation of the type which doesn't do anything.